iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -28,6 +30,46 @@ namespace iree_compiler {
 
 namespace {
 
+LogicalResult sanitizeSymbolNames(IREE::VM::ModuleOp &moduleOp) {
+  auto replaceIllegalChars = [](std::string &symbolName) {
+    std::regex illegalCharactersRe("<|>");
```

Comment:
I think regex is a bit overkill for this given that it's just 3 characters; not that this particular instance is the worst as it's once per module, but it's often a good idea not to let stuff like this leak in as someone else may take it as inspiration for something called 10000 times :)

If you think there will be many more characters or something you'd actually want to use the capabilities of regex for (like matching entire character classes) that's fine, but then you can replace this entire thing with a use of `std::regex_replace` that does the same thing in 1 line.

Reply:
(actually just caught that it is per function/global/etc - still relatively small in the big scheme, but something to watch out for in the future when reaching for regex nonetheless)

Reply:
As for the regex, you probably want something like the inverse of `/^[a-zA-Z_][a-zA-Z0-9_]*$/` (I'm sure there are better examples floating around on the net) - otherwise the next time a non-alphanumeric character gets tried it'll fail.

Reply:
I aggree, it is overkill :) Originally, there were two calls to `std::replace` which I thought was also not really nice. And there was the problem, that I couldn't figure out if the string was actually changed - a string compare would have worked but that seemed a little odd. I used iterator approach because it provided me with exactly this ability. AFAIK you cannot tell if there was a match with `std::regex_replace` :/
Changing the regex makes sense. Let's see with what I can come up with.

Reply:
I think all of this could be done with a for loop around `c == '_' || llvm::isAlnum(c)` (which is basically just isalpha || isdigit)  and read much better.

Something like:
```c
auto replaceIllegalChars = [](std::string &str) {
  char* ptr = const_cast<char*>(str.data());
  for (size_t i = 0; i < str.size(); ++i) {
    if (ptr[i] != '_' && !llvm::isAlnum(ptr[i])) ptr[i] = '_';
  }
};
```
And for good measure, prefixing the symbol names emitted so you don't even have to worry about leading digits and such.

There's literally 3 character classes we can allow in a C identifier and everything else needs to be replaced. Keeping these things simple and understandable when they're this constrained is good practice as ideally with something conservative like the above no one will ever have to look at this code again - and if they do they don't need to try to work out the regex rules and iterators and other goo :)

Reply:
I like the simplicity of your approach! Everyone understands it right away - nice!